lib/datadog

```diff
@@ -0,0 +1,41 @@
+function install_datadog() {
+    status "Datadog usage detected, installing PHP extensions"
+
+    local tracer_version="${1}"
+    local appsec_version="${2}"
+
+    if [ "${tracer_version}" = "0" ] && [ "${appsec_version}" = "0" ]; then
+        echo >&2 "Skipping Datadog installation, as there is nothing to install."
+    fi
+
+    local tracer_option
+    if [ "${tracer_version}" = "0" ]; then tracer_option="--no-tracer"; else tracer_option="--version=${tracer_version}"; fi
+
+    local appsec_option
+    if [ "${appsec_version}" = "0" ]; then appsec_option="--no-appsec"; else appsec_option="--version=${appsec_version}"; fi
+
+    local install_dir="${BUILD_DIR}/vendor/datadog"
+    local cwd=$(pwd)
+    local temp_dir=$(mktmpdir "datadog")
+
+    cd "${tempdir}"
+
+    # the AppSec extension is in public beta right now, and the URL of the installer might change in the future
+    curl --silent -L "https://raw.githubusercontent.com/DataDog/dd-appsec-php/installer/dd-library-php-setup.php" > dd-library-php-setup.php
```

Comment:
I have added a comment on the previous line, because the Datadog Appsec feature is still in public beta and the URL of the installer might change in the future. It currently targets a file that is not in the main branch, see https://github.com/DataDog/dd-appsec-php/issues/75